src/Microsoft.OData.Core/JsonLight/ODataJsonLightContextUriParser.cs

```diff
@@ -71,19 +71,44 @@ private ODataJsonLightContextUriParser(IEdmModel model, Uri contextUriFromPayloa
             ODataPayloadKind payloadKind,
             Func<IEdmType, string, IEdmType> clientCustomTypeResolver,
             bool needParseFragment,
-            bool throwIfMetadataConflict = true)
+            bool throwIfMetadataConflict = true,
+            Uri baseUri = null,
+            string navigationSource = null)
         {
             if (contextUriFromPayload == null)
             {
                 throw new ODataException(ODataErrorStrings.ODataJsonLightContextUriParser_NullMetadataDocumentUri);
             }
 
-            // Create an absolute URI from the payload string
-            // TODO: Support relative context uri and resolving other relative uris
+            // Create a context URI from the payload string.
             Uri contextUri;
             if (!Uri.TryCreate(contextUriFromPayload, UriKind.Absolute, out contextUri))
             {
-                throw new ODataException(ODataErrorStrings.ODataJsonLightContextUriParser_TopLevelContextUrlShouldBeAbsolute(contextUriFromPayload));
+                if (baseUri != null)
+                {
+                    ODataUri oDataUri = new ODataUri() { ServiceRoot = baseUri };
+
+                    // Graph requires this format for delta request body context.

```

Comment:
Does it mean this code caters to MS Graph's unique requirement? Can it be generalized such that it will work for different categories of customers?

Reply:
Or did you mean not including that comment?